docs/reference/mapping/runtime.asciidoc

```diff
@@ -0,0 +1,518 @@
+        }
+      }
+    },
+    "properties": {
+      "timestamp": {"type": "date"}
+    }
+  }
+}
+----
+
+<1> Runtime fields are defined in the `"runtime"` section of the mapping
+definition.
+<2> Each runtime has its own field type, just like any other field.
+<3> The script defines the evaluation to calculate at search time.
+
+The `"runtime"` section supports `boolean`, `date`, `double`, `ip`, `keyword`,
+and `long` data types. Runtime fields with a `type` of `date` can accept the
+<<mapping-date-format,`format`>> parameter exactly as the `date` field type.
+
+IMPORTANT: Objects are not supported in the `"runtime"` section under the
+mapping definition. If you want to map a runtime field under an object, you
+can
+<<runtime-shadowing-fields,shadow the field value in the search request>>.
+
+[[runtime-updating-scripts]]
+.Updating runtime scripts
+****
+
+Updating a script while a dependent query is running can return
+inconsistent results. Each shard might have access to different versions of the
+script, depending on when the mapping change takes effect.
+
+Existing queries or visualizations in {kib} that rely on runtime fields can
+fail if you change the field type. For example, a bar chart visualization
+that uses a runtime field of type `ip` will fail if the type is changed
+to `boolean`.
+
+****
+
+[discrete]
+[[runtime-search-request]]
+=== Defining runtime fields in a search request
+You can specify a `runtime_mappings` section in a search request to create
+runtime fields that exist only as part of the query. You specify a script
+as part of the `runtime_mappings` section, just as you would if adding a
+runtime field to the mappings.
+
+In the following request, the values for the `day_of_week` field are calculated
+dynamically, and only within the context of this search request:
+
+[source,console]
+----
+GET my-index/_search
+{
+  "runtime_mappings": {
+    "day_of_week": {
+      "type": "keyword",
+      "script": {
+        "source": "emit(doc['@timestamp'].value.dayOfWeekEnum.getDisplayName(TextStyle.FULL, Locale.ROOT))"
+      }
+    }
+  },
+  "aggs": {
+    "day_of_week": {
+      "terms": {
+        "field": "day_of_week"
+      }
+    }
+  }
+}
+----
+// TEST[continued]
+
+[discrete]
+[[runtime-shadowing-fields]]
+==== Shadowing fields in the search request
+If you create a runtime field with the same name as a field that
+already exists in the mapping, the runtime field shadows the mapped field. At
+search time, {es} calculates the value of the runtime field and returns it as
+part of the query. Because the runtime field shadows the mapped field, you can
+modify the value returned in search without modifying the mapped field.
+
+Objects are not supported in the `"runtime"` section under the
+mapping definition. However, you can use the `runtime_mappings` section of the
+`_search` request to shadow field values inside objects. For
```

Comment:
similar comment as above. Does it sound like a limitation? Defining fields that belong to an object, using the dot notation, does not necessarily mean that you are shadowing an existing field. It's two separate things.

Reply:
I removed the mention about objects not being supported after our discussion.